feature-libs/product-configurator/rulebased/root/interceptor/cpq-configurator-rest.interceptor.ts

```diff
@@ -39,7 +39,7 @@ export class CpqConfiguratorRestInterceptor implements HttpInterceptor {
           catchError((errorResponse: any) => {
             return this.handleError(errorResponse, next, request);
           }),
-          tap((response) => this.extractCpqSessionId(response, cpqData))
+          tap((response) => this.extractCpqSessionId(response))
```

Comment:
@Uli-Tiger Do I get the flow correct?:
1. when you any call to CPQ server, we intercept it
1. we enrich it with the cpq session ID header (which is null at the beginning)
1. in case of 403 error, we renew the token (which is likely the case for the first call)
1. when it's renewed, we retry the same call again in the method `handleError()`, **but we don't extract the cpq session ID there**, so it's **still null**
1. later on when we make a brand new call to CPQ server, we enrich it with **null** cpq session ID
1. only when we get the response from the succesfull call, we extract the cpq session ID

Do I understand well? If yes, is it correct behavior (especially we don't extract session ID in point 4.)

Reply:
>     3. in case of 403 error, we renew the token (which is likely the case for the first call)

no - you will always get a fresh token for the first call. It's requested from CPQ via commerce (as only the commerce server may request client tokens, but not the client itself) Client tokens are NOT cached server side.
So the 403 will only happen if the token really expires. Whether this happens seldom or regular, depends on token lifetime. Currently its 24 hours.
> when it's renewed, we retry the same call again in the method handleError(), but we don't extract the cpq session ID there, so this.cpqSessionId is still null
I verified in postman: In case of a 403 no CPQ Session id is returned. This also makes sense, as the request is rejected before any session handling can happen. So no need to extract a session id.